### PR TITLE
Implement new design for tooltips in GUI

### LINKED ISSUE #10088: Restyle tooltips to new deisgn
image

https://www.figma.com/design/MXE4HsnOgBwI49aXg0byIg/ide2-v124?node-id=0-1&t=gi4VgWRM34BxUGK3-1

### LINKED ISSUE #10154: CB breadcrumbs have a blank space on the left
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [X] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

Screenshot 2024-06-03 at 12 30 05 PM

### Expected behaviour

I guess back/forward arrows are missing there?

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Logs

_No response_

### Enso Version

6adf590d9f7dd5080291a5bcc2752ea55338f832

### Browser or standalone distribution

Standalone distribution (local project)

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by Frizi**
### Pull Request Description

Fixes #10088, #10154

Implemented figma tooltip design through Vue components and store. Attached tooltips to all existing SVG buttons, based on assigned label property.

https://github.com/enso-org/enso/assets/919491/85f5fef7-c6df-471b-b544-b45f45f4f51e

### Important Notes

Removed all usages of `v-bind` in styles due to issues during hot reloading when updating stores. The internal CSS binding composable sometimes crashes when the component is being unmounted due to hot reload.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by somebody1234 — APPROVED** [review]
Approving changes to `RemoteBackend.ts` and `projectManagerShimMiddleware.ts`

File: app/gui2/src/asyncApp.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import '@/assets/main.css'
 
 export async function AsyncApp() {
+  console.log('AsyncApp')
```
Comment (vitvakatu) [inline app/gui2/src/asyncApp.ts]: Left over console.log?

File: app/gui2/src/components/GraphEditor.vue [hunk 2]
```diff
@@ -73,12 +83,16 @@ const _visualizationStore = provideVisualizationStore(projectStore)
 
 widgetRegistry.loadBuiltins()
 
+console.log('==Graph setup==')
```
Comment (vitvakatu) [inline app/gui2/src/components/GraphEditor.vue]: 3 console.logs in this file

File: app/gui2/src/components/MenuButton.vue [hunk 3]
```diff
@@ -9,23 +11,33 @@
  * `disabled` class.
  */
 
+// defineOptions({ inheritAttrs: false })
```
Comment (vitvakatu) [inline app/gui2/src/components/MenuButton.vue]: commented code

File: app/gui2/src/components/ScrollBar.vue [hunk 4]
```diff
@@ -100,28 +100,27 @@ const clickTrack = {
...
-  | {
-      type: 'jump'
-      axis: 'x' | 'y'
-      position: number
-    }
+  | { type: 'start' }
+  | { type: 'move'; startOffset: Vec2 }
+  | { type: 'stop' }
+  | { type: 'jump'; axis: 'x' | 'y'; position: number }
+export default {}
```
Comment (vitvakatu) [inline app/gui2/src/components/ScrollBar.vue:119]: Just wondering, why is it needed?

File: app/gui2/src/composables/syncLocalStorage.ts [hunk 5]
```diff
@@ -75,6 +75,9 @@ export function useSyncLocalStorage<StoredState extends Object>(
 
   let nextRestoreId = 0
   const restoreIdInProgress = ref<number>()
+  ;(restoreIdInProgress as any).__DEBUG = true
+  console.log('restoreIdInProgress', restoreIdInProgress)
```
Comment (vitvakatu) [inline app/gui2/src/composables/syncLocalStorage.ts]: Some debug code

File: app/gui2/src/composables/syncLocalStorage.ts [hunk 6]
```diff
@@ -119,17 +122,13 @@ export function useSyncLocalStorage<StoredState extends Object>(
 
   async function restoreState(storageKey: string) {
     abortLastRestore()
-    assert(restoreIdInProgress.value == null)
+    // assert(restoreIdInProgress.value == null)
```
Comment (vitvakatu) [inline app/gui2/src/composables/syncLocalStorage.ts]: Commented assert

File: app/gui2/src/util/patching.ts [hunk 7]
```diff
@@ -11,3 +13,57 @@ export function hookBeforeFunctionCall(object: any, key: PropertyKey, hook: () =
...
+ * Force propagation of parent scopes into all root nodes of this component, i.e. children of it's
+ * top-level Fragment VNode.
+ *
+ * By default, Vue is restricting scope propagation to only happen within components with only
+ * singular root node. Unfortunately, that also happens to block propagation when components adds an
+ * additional root-node, such as "Teleport". This is a workaround that tricks Vue into actually
+ * propagating the scopes past this component's top-level "Fragment" element, allowing parent scoped
+ * styles to affect its actual root node or rendered slots.
+ */
+export function usePropagateScopesToAllRoots() {
```
Comment (vitvakatu) [inline app/gui2/src/util/patching.ts:27]: I’m speechless 🤯

**Review by vitvakatu — APPROVED** [review]
Minor comments

File: app/gui2/src/components/ScrollBar.vue (hunk 4 above)
Comment (Frizi) [reply app/gui2/src/components/ScrollBar.vue:119]: It doesn't do anything, really. Exporting default from script allows you to specify extra component options, but in this case it is empty anyway. This change only fixes an issue with code highlighting in the editor. For some reason, when the block has only a union type, VSCode fails to properly detect the closing tag and wrongly colors the whole template/style below.

**Review by kazcw — APPROVED** [review]
Since `usePropagateScopesToAllRoots` depends on Vue internals, it would be nice to have an automated test in case a minor update breaks it.

**Comment by Frizi** [comment]
Creating this test is proving to be quite tricky. I'll add it as a separate PR to not block merging this, since I already have another PR to do that will depend on it.

**Pull request merged**